### PR TITLE
feat: add support for parallel backtests

### PR DESCRIPTION
**Pull request by Caceresenzo**

### CODE REVIEW

File: bktest/iterator.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+from .export import ExporterCollection
+
+
+@dataclasses.dataclass()
+class Postpone:
+    
+    date: datetime.date
+    reason: str
+    ordered: bool
```
Comment (matteoettam09) [inline bktest/iterator.py:13]: Ordered always true here.

File: bktest/backtest.py [hunk 2]
```diff
@@ -1,86 +1,43 @@
...
-
-    def order(self, date: datetime.date, price_date=None) -> _MassOrderResult:
-        mass_result = _MassOrderResult(order_results=[])
+    def order(
+        self,
+        date: datetime.date,
+        orders: typing.List[Order],
+        price_date=None,
+    ) -> OrderResultCollection:
+        mass_result = OrderResultCollection()
```
Comment (matteoettam09) [inline bktest/backtest.py:36]: please rename

**Review by matteoettam09 — COMMENTED** [review]
gg

**Review by matteoettam09 — CHANGES_REQUESTED** [review]
see Comments above.

File: bktest/iterator.py (hunk 1 above)
Comment (Caceresenzo) [reply bktest/iterator.py]: fixed via https://github.com/crunchdao/backtest/pull/25/commits/9a651caa36c7e8f5c7b45bd97bd3362d7ab390e9

File: bktest/backtest.py (hunk 2 above)
Comment (Caceresenzo) [reply bktest/backtest.py]: https://github.com/crunchdao/backtest/pull/25/commits/bd26e36d7d8b37628f104d4ef9b37c17b7ca2705

**Review by xaviergilbert — APPROVED** [review]

**Pull request merged**